### ISSUE TITLE
[FEA] Audit_3.0.1: Minor change in BatchScanExec

### ISSUE THREAD

**Issue by razajafri** [NVIDIA/spark-rapids#957]
commit 30e3fcbb081b6b07a12316c9dd63c66f77b8fec7
Author: Ryan Blue <blue@apache.org>
Date:   Wed Jul 8 16:06:40 2020 -0700

    [SPARK-32168][SQL] Fix hidden partitioning correctness bug in SQL overwrite

**Comment by kuhushukla** [comment]
@razajafri  Could we elaborate on why we need to audit this further?

**Comment by nartal1** [comment]
In BatchScanExec, this is the change.
```
-  override lazy val partitions: Seq[InputPartition] = batch.planInputPartitions()
+  @transient override lazy val partitions: Seq[InputPartition] = batch.planInputPartitions()
```
GpuBatchScan needs to be updated.

**Comment by jlowe** [comment]
Fixed by #5897
